packages/terra-worklist-data-grid/src/subcomponents/Cell.jsx

```diff
@@ -0,0 +1,279 @@
+   */
+  acceptsFocus: PropTypes.bool,
+
+  /**
+   * Boolean indicating whether the Cell is actively selected.
+   */
+  isSelected: PropTypes.bool,
+
+  /**
+   * @private
+   * Boolean indicating whether the cell is a row selection cell.
+   */
+  isRowSelectionCell: PropTypes.bool,
+
+  /**
+   * String that labels the cell for accessibility.
+   */
+  ariaLabel: PropTypes.string,
+
+  /**
+   * Class name for the cell.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Function that is called when a selectable cell is selected. Parameters: `onCellSelect(rowId, columnId)`
+   */
+  onCellSelect: PropTypes.func,
+
+  onCellSelectionChange: PropTypes.func,
+  onMoveCellFocus: PropTypes.func,
+  onNavigationModeChange: PropTypes.func,
+
+  isRowSelectionModeEnabled: PropTypes.bool,
+  isRowHeader: PropTypes.bool,
+  isNavigationEnabled: PropTypes.bool,
+
+  /**
+   * Content that will rendered within the Cell.
+   */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  acceptsFocus: false,
+  isRowSelectionCell: false,
+  isRowHeader: false,
+};
+
+function Cell(props) {
+  const {
+    rowId,
+    columnId,
+    coordinates,
+    acceptsFocus,
+    isSelected,
+    isRowSelectionCell,
+    ariaLabel,
+    className,
+    onCellSelectionChange,
+    onCellSelect,
+    onMoveCellFocus,
+    isRowSelectionModeEnabled,
+    onNavigationModeChange,
+    isRowHeader,
+    children,
+    isNavigationEnabled,
+    ...customProps
+  } = props;
+
+  const selectCell = (event) => {
+    // If current cell is selected, do nothing.
+    if (isSelected) {
+      event.preventDefault();
+      return;
+    }
+
+    // Make note of cell that is currently selected.
+    if (onCellSelectionChange) {
+      onCellSelectionChange(rowId, columnId, coordinates);
+    }
+
+    event.preventDefault();
+
+    if (onCellSelect) {
+      onCellSelect(rowId);
+    }
+  };
+
+  const handleClick = (event) => {
+    if (isRowSelectionModeEnabled) {
+      if (onCellSelectionChange) {
+        onCellSelectionChange(null, null, coordinates);
+      }
+    } else {
+      selectCell(event);
+    }
+  };
+
+  const handleKeyUp = (event) => {
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_SPACE:
+        if (isRowSelectionModeEnabled) {
+          if (onCellSelectionChange) {
+            onCellSelectionChange(null, null, coordinates);
+          }
+        } else {
+          selectCell(event);
+        }
+        event.preventDefault();
+        break;
+      default:
+    }
+  };
+
+  const handleKeyDownWhenGridNavigationDisabled = (event) => {
+    // TODO: Set focus to the first focusable child of the active cell.
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_ESCAPE:
+        if (onNavigationModeChange) {
+          onNavigationModeChange(true);
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleKeyDown = (event) => {
```

Comment:
The WorklistDataGrid would handle the key events accept for the one(s) that select the cell.